tutorials/migrating-rails-app-from-heroku-to-google-appengine.md

```diff
@@ -0,0 +1,418 @@
+it if your application boots differently:
+
+    entrypoint: bundle exec rails s -p $PORT
+
+App Engine provides the `$PORT` environment variable for you, to tell your app
+which port it should listen on. (Usually it is set to 8080.) Many web
+frameworks automatically honor this environment variable, but it is good
+practice to reference it explicitly in your entrypoint.
+
+### Configure environment variables and secrets
+
+Heroku provides a way to manage configuration via
+[config vars](https://devcenter.heroku.com/articles/config-vars). It injects
+them as environment variables into your app's runtime environment.
+
+App Engine provides two ways to set environment variables, depending on how
+you'd like to manage and secure the content. First, you can set them directly
+in the `app.yaml` file by creating an `env_variables` section, which App Engine
+injects into your app's runtime environment. For example:
+
+    env_variables:
+      DATABASE_USER: postgres
+      RAILS_MAX_THREADS: "5"
+
+This is useful for simple configuration. However, you typically want to
+save your `app.yaml` config file to source control, so it is not a good place
+for secrets such as your database password or your `SECRET_KEY_BASE`. This
+kind of sensitive data can instead be managed securely using the
+[Runtime Configuration Service](https://cloud.google.com/deployment-manager/runtime-configurator/),
+a Google service similar to the Heroku config system.
+
+In your app, you likely need to manage at least two secret values: the
+`DATABASE_PASSWORD`, and the `SECRET_KEY_BASE`. Let's store them in Runtime
+Configuration:
+
+1. [Enable the Runtime Config API in the Cloud Console](https://console.cloud.google.com/apis/library/runtimeconfig.googleapis.com).
+
+1.  Create a configuration named `my-env` by running the following command:
+
+        gcloud beta runtime-config configs create my-env
+
+1.  View the config values for your Heroku app:
+
+        heroku config
+
+    Copy the `SECRET_KEY_BASE` value so you can paste it in the next command.
+
+1.  Set the `SECRET_KEY_BASE` in your Google Runtime Config:
+
+        gcloud beta runtime-config configs variables set \
+          --config-name=my-env --is-text \
+          SECRET_KEY_BASE "VeryLongStringCopiedFromHeroku"
+
+1.  Previously you set up your `database.yml` configuration to read from the
+    `DATABASE_PASSWORD` environment variable. Set that up:
+
+        gcloud beta runtime-config configs variables set \
+          --config-name=my-env --is-text \
+          DATABASE_PASSWORD [MY_DATABASE_PASSWORD]
+
+    Replace `${MY_DATABASE_PASSWORD}` with the password that you used to secure
+    your database.
+
+If you have any other sensitive configuration fields, copy them over in the
+same way.
+
+Next, instruct your application to load this configuration into
+environment variables.
+
+1.  First, add the following to your `app.yaml`:
+
+        runtime_config:
+          dotenv_config: my-env
```

Comment:
Oh? is this available now? 🎆

Reply:
Yep it is. I do still need to document it. :)